include/zephyr/drivers/adc/adc_current_sense_shunt.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023 The ChromiumOS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_ADC_CURRENT_SENSE_SHUNT_H_
+#define ZEPHYR_INCLUDE_DRIVERS_ADC_CURRENT_SENSE_SHUNT_H_
+
+#include <zephyr/drivers/adc.h>
+
+struct adc_dt_current_sense_shunt_cfg {
```

Comment:
Looking at other structures we have like `adc_dt_spec`, `gpio_dt_spec`, `pwm_dt_spec`. Can you follow that pattern? This could be `current_sense_dt_spec`. Same for the initializer (`ADC_DT_SPEC_GET`, `ADC_DT_SPEC_GET_BY_IDX` for reference)